research-software-directory/RSD-as-a-service
# Better OpenAlex support

## Better OpenAlex support

### Changes proposed in this pull request

* Allow for finding mentions by OpenAlex ID
* Also search in the OpenAlex database when searching for mentions by title
* Expand the mentions scraper to also include mentions with an OpenAlex ID (and no DOI)
* Updated the respective docs

### How to test

* `docker compose down --volumes && docker compose build --parallel && docker compose up --scale data-generation=0`
* Add mentions with a DOI
* Add mentions with an OpenAlex ID but no DOI (see e.g.https://research-software-directory.org/api/v1/mention?external_id=not.is.null&doi=is.null&select=title,doi,external_id&limit=100)
* Add mentions by searching for a title, OpenAlex results should also show up
* Add these mentions in various places (reference papers for software, other mentions for software, project impact, project output)
* Afterwards, run the mentions scraper, no errors should be reported:
* `docker compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.doi.MainMentions`


Closes #1312

PR Checklist:
